lib/common/result/result.dart

```diff
@@ -1,15 +1,18 @@
-import 'package:flutter/foundation.dart';
+import 'dart:async';
 
-/// Result class to handle success, failure and loading states.
+import 'package:chopper/chopper.dart';
+import 'package:dcc_toolkit/chopper/base_error.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:http/http.dart';
+
+/// Result class to handle success and failure states.
 @immutable
 sealed class Result<T> {
   const Result();
 
   factory Result.success(T value) => Success(value);
 
-  factory Result.failure(Exception? exception) => Failure(exception);
-
-  factory Result.loading() => Loading<T>();
+  factory Result.failure(BaseError? exception) => Failure(exception);
```

Comment:
Did we not want to use `Object?` so that we're not bound to any specific error implementation